sign/ed25519/ed25519.go

```diff
@@ -52,15 +76,32 @@ func (kp *KeyPair) Seed() []byte { return kp.GetPrivate() }
 func (kp *KeyPair) Public() crypto.PublicKey { return kp.GetPublic() }
 
 // Sign creates a signature of a message given a key pair.
-// Ed25519 performs two passes over messages to be signed and therefore cannot
-// handle pre-hashed messages.
+// This function can handle unhashed messages or messages that will be
+// prehashed with SHA512. It also handles context.
 // The opts.HashFunc() must return zero to indicate the message hasn't been
 // hashed. This can be achieved by passing crypto.Hash(0) as the value for opts.
+// The opts.HashFunc() must return SHA512 to indicate that the message will be
+// hashed with SHA512. This can be achieved by passing crypto.SHA512 as the value
+// for opts.
+// Messages to be prehashed with other algorithms are not handled.
```

Comment:
Replace the doc string by this shorter one.

```suggestion
// This function supports all the three signature variants defined in RFC-8032, namely Ed25519 (or pure EdDSA), Ed25519Ph, and Ed25519Ctx.
// The context is passed inside a Options struct defined in this package.
```

Reply:
I think specifying Hash(0) or SHA-512 inside ops is useless, since the hash function is fixed.
However, let's keep it (in this function) as is. And let's follow how upstream stdlib solves this API.